File: app/src/main/java/com/ywwynm/everythingdone/utils/StringUtil.java
package com.ywwynm.everythingdone.utils;

import android.text.TextUtils;

/**
 * Created by ywwynm on 2016/10/20.
 * some utils to operate String
 */
public class StringUtil {

    public static final String TAG = "StringUtil";

    private StringUtil() {}

    public static String upperFirst(String s) {
        if (TextUtils.isEmpty(s)) {
            return "";
        }
        String first = String.valueOf(s.charAt(0));
        return first.toUpperCase() + s.substring(1, s.length());
    }

    public static String lowerFirst(String s) {
        if (TextUtils.isEmpty(s)) {
            return "";
        }
        String first = String.valueOf(s.charAt(0));
        return first.toLowerCase() + s.substring(1, s.length());
    }

    public static String replaceChineseBrackets(String s) {
        if (TextUtils.isEmpty(s)) {
            return "";
        }
        return s.replaceAll("（", "(").replaceAll("）", ")");
    }

}

File: app/src/main/java/com/ywwynm/everythingdone/utils/StringUtil.kt
package com.ywwynm.everythingdone.utils

import android.text.TextUtils

import java.util.Locale

/**
 * Created by ywwynm on 2016/10/20.
 * Translated to Kotlin by ywwynm and Claude Opus 4.7 on 2026/5/20.
 * some utils to operate String
 */
object StringUtil {

    const val TAG: String = "StringUtil"

    @JvmStatic
    fun upperFirst(s: String?): String {
        if (TextUtils.isEmpty(s)) {
            return ""
        }
        val first: String = s!![0].toString()
        return first.uppercase(Locale.getDefault()) + s.substring(1, s.length)
    }

    @JvmStatic
    fun lowerFirst(s: String?): String {
        if (TextUtils.isEmpty(s)) {
            return ""
        }
        val first: String = s!![0].toString()
        return first.lowercase(Locale.getDefault()) + s.substring(1, s.length)
    }

    @JvmStatic
    fun replaceChineseBrackets(s: String?): String {
        if (TextUtils.isEmpty(s)) {
            return ""
        }
        return s!!.replace("（".toRegex(), "(").replace("）".toRegex(), ")")
    }

}
